meta/saimetadatatypes.h

```diff
@@ -53,6 +53,13 @@ typedef struct _sai_object_meta_key_t
      */
     sai_object_key_t            objectkey;
 
+    /**
+     * @brief The key extensions.
+     *
+     * @passparam objecttype
+     */
+    sai_object_key_extension_t  objectkeyext;
```

Comment:
this seems to me like a bad idea, you still will need to include extensions header to support those new experimental object types, why not include them in sai_object_key_t, if you include experimental in existing one, you will not have to modify any metadata utils or parser s